core/src/main/java/bisq/core/dao/burningman/BurningManPresentationService.java

```diff
@@ -0,0 +1,322 @@
+                        .map(genesisTransaction -> genesisTransaction.getOutputs().stream()
+                                .filter(transactionOutput -> transactionOutput.isMine(bsqWalletService.getWallet()))
+                                .map(transactionOutput -> BurningManService.GENESIS_OUTPUT_PREFIX + transactionOutput.getIndex())
+                                .collect(Collectors.toSet())
+                        )
+                );
+        return myGenesisOutputNames;
+    }
+
+    public Set<String> getMyCompensationRequestNames() {
+        // Can be empty, so we compare with null and reset to null at new block
+        if (myCompensationRequestNames != null) {
+            return myCompensationRequestNames;
+        }
+        myCompensationRequestNames = myProposalListService.getList().stream()
+                .filter(proposal -> proposal instanceof CompensationProposal)
+                .map(Proposal::getName)
+                .collect(Collectors.toSet());
+        return myCompensationRequestNames;
+    }
+
+    public Map<String, BurningManCandidate> getBurningManCandidatesByName() {
+        // Cached value is only used for currentChainHeight
+        if (!burningManCandidatesByName.isEmpty()) {
+            return burningManCandidatesByName;
+        }
+
+        burningManCandidatesByName.putAll(burningManService.getBurningManCandidatesByName(currentChainHeight));
+        return burningManCandidatesByName;
+    }
+
+    public LegacyBurningMan getLegacyBurningManForDPT() {
+        if (legacyBurningManDPT.isPresent()) {
+            return legacyBurningManDPT.get();
+        }
+
+        // We do not add the legacy burningman to the list but keep it as class field only to avoid that it
+        // interferes with usage of the burningManCandidatesByName map.
+        // Those are the hashes used by legacy BM for burning BTC received from DPT.
+        Set<String> hashes = Set.of("1701e47e5d8030f444c182b5e243871ebbaeadb5e82f",
+                "1701293c488822f98e70e047012f46f5f1647f37deb7");
+        LegacyBurningMan legacyBurningManDPT = getLegacyBurningMan(burningManService.getLegacyBurningManAddress(currentChainHeight), hashes);
+
+        this.legacyBurningManDPT = Optional.of(legacyBurningManDPT);
+        return legacyBurningManDPT;
+    }
+
+    public LegacyBurningMan getLegacyBurningManForBtcFees() {
+        if (legacyBurningManBtcFees.isPresent()) {
+            return legacyBurningManBtcFees.get();
+        }
+
+        // We do not add the legacy burningman to the list but keep it as class field only to avoid that it
+        // interferes with usage of the burningManCandidatesByName map.
+        // The hash used by legacy BM for burning BTC received from BTC trade fees.
+        Set<String> hashes = Set.of("1701721206fe6b40777763de1c741f4fd2706d94775d");
+        LegacyBurningMan legacyBurningManBtcFees = getLegacyBurningMan(LEGACY_BURNING_MAN_BTC_FEES_ADDRESS, hashes);
```

Comment:
Same again, hardcoded?  I have to replace it with the relevant hashes from my database in order to pick up the legacy burningman's activity.

Reply:
You mean for testing it with regtest?

Reply:
Yes, Regtest.  I'm confused as to the necessity for hardcoding those hashes: if BM3 does a fee burn, I would have expected it to get integrated in real time, instead of pasting into the sourcecode.

Reply:
The burn tx opReturn data are arbitrary bytes. One can add some preimage which gets hashed in the UI, and that is what the BM is doing to tag the diff. burn txs to groups. So this helps us to filter for burn txs from btc fees and those for DPTs.